scripts/jira_tickets/create_healthcheck_tickets.pl

```diff
@@ -102,34 +66,34 @@ =head2 SUMMARY
 #----------------------------------#
 
 # create initial ticket for HC run - failures will become subtasks of this
-my $blocked_ticket_key = find_handover_ticket($release, $division, $parameters, $logger);
-my $hc_task_json_ticket = {
-    assignee    => $parameters->{user},
+my $blocked_ticket_key = find_handover_ticket($jira_adaptor);
+my $hc_task_json_ticket = [{
+    assignee    => $jira_adaptor->{_user},
     summary     => "$hc_basename ($timestamp)",
     description => "Java healthcheck failures for HC run on $timestamp\nFrom file: $hc_abs_path",
     links       => [ ['Blocks', $blocked_ticket_key] ],
-};
-my $hc_task_jira_ticket = Bio::EnsEMBL::Compara::Utils::JIRA->json_to_jira($hc_task_json_ticket, 'Task', $parameters, $logger);
+}];
 
 # create subtask tickets for each HC failure
-my @jira_subtasks;
+my @json_subtasks;
 foreach my $testcase ( keys %$testcase_failures ) {
     my $failure_subtask_json = {
         summary => $testcase,
         description => $testcase_failures->{$testcase},
     };
-    push(@jira_subtasks, Bio::EnsEMBL::Compara::Utils::JIRA->json_to_jira($failure_subtask_json, 'Sub-task', $parameters, $logger));
-}
-
-# POST tickets to JIRA
-my $hc_task_ticket_key = Bio::EnsEMBL::Compara::Utils::JIRA->create_ticket( $hc_task_jira_ticket, $parameters, $logger );
-print "Created ticket: $hc_task_ticket_key\n";
-create_blocker_link( $hc_task_ticket_key, $blocked_ticket_key );
-foreach my $subtask ( @jira_subtasks ) {
-    $subtask->{'fields'}->{'parent'} = { 'key' => $hc_task_ticket_key };
-    my $subtask_ticket_key = Bio::EnsEMBL::Compara::Utils::JIRA->create_ticket( $subtask, $parameters, $logger );
-    print "\tSubtask: $subtask_ticket_key\n";
+    push(@json_subtasks, $failure_subtask_json);
 }
+# Add subtasks to the initial ticket
+$hc_task_json_ticket->[0]->{subtasks} = \@json_subtasks;
+my $components = ['Java Healthchecks', 'Production tasks'];
+# Create all JIRA tickets
+my $hc_task_keys = $jira_adaptor->create_tickets(
+    -JSON_INPUT => encode_json($hc_task_json_ticket), -PRIORITY => 'Blocker', -COMPONENTS => $components,
+    -DRY_RUN => $dry_run
+);
+# Create a blocker issue link between the newly created HC ticket and the
+# handover ticket
+$jira_adaptor->link_tickets('Blocks', $hc_task_keys->[0], $blocked_ticket_key, $dry_run);
```

Comment:
I forgot to remove the `links` key from the previous code. It needs to be done after because JIRA wouldn't add the second part of the link in that case (`$blocked_ticket_key` blocked by `hc_task_json_ticket`).